yt_dlp/extractor/line.py

```diff
@@ -82,15 +84,50 @@ def _real_extract(self, url):
 
         # like_count requires an additional API request https://tv.line.me/api/likeit/getCount
 
+        thumbnail = url_or_none(try_get(video_info, lambda x: x['meta']['cover']['source']))
+        if thumbnail:
+            thumbnail = thumbnail.split('?')[0]
+
+        subtitles = {}
+        sub_fan = "-fan"
+        sub_dict = try_get(video_info, lambda x: x['captions']['list']) or {}
+
+        for sub_index in sub_dict:
+            sub_language = try_get(sub_index, lambda x: x['language'])
+            sub_source = try_get(sub_index, lambda x: x['source'])
+            sub_label = try_get(sub_index, lambda x: x['label'])
+            sub_country = try_get(sub_index, lambda x: x['country'])
+            # Checking for fan provided subtitles in already provided languages
+            if subtitles:
+                for subtitle in subtitles.keys():
+                    if subtitle == sub_language + "-" + sub_country or subtitle == sub_language + "-" + sub_country + sub_fan:
+                        sub_lang = sub_language + "-" + sub_country + sub_fan
+                        sub_name = sub_label + sub_fan
+                    else:
+                        sub_lang = sub_language + "-" + sub_country
+                        sub_name = sub_label
+            else:
+                sub_lang = sub_language + "-" + sub_country
+                sub_name = sub_label
```

Comment:
Use fstrings

Reply:
```suggestion
                    if subtitle in (f'{sub_language}-{sub_country}', f'{sub_language}-{sub_country}{sub_fan}'):
                        sub_lang = f'{sub_language}-{sub_country}{sub_fan}'
                        sub_name = f'{sub_label}{sub_fan}'
                    else:
                        sub_lang = f'{sub_language}-{sub_country}'
                        sub_name = sub_label
            else:
                sub_lang = f'{sub_language}-{sub_country}'
                sub_name = sub_label
```
Could probably be improved further to reduce repetition.

Reply:
```suggestion
                    if sub_language in subtitle:
                        if sub_country:
                            sub_lang = f'{sub_language}-{sub_country}{sub_fan}'
                        else:
                            sub_lang = f'{sub_language}{sub_fan}'
                        sub_name = f'{sub_label}{sub_fan}'
                    else:
                        if sub_country:
                            sub_lang = f'{sub_language}-{sub_country}'
                        else:
                            sub_lang = sub_language
                        sub_name = sub_label
            else:
                if sub_country:
                    sub_lang = f'{sub_language}-{sub_country}'
                else:
                    sub_lang = sub_language
                sub_name = sub_label
```